src/main/kotlin/nl/info/zac/policy/PolicyService.kt

```diff
@@ -0,0 +1,188 @@
+import nl.info.client.zgw.ztc.model.generated.StatusType
+import nl.info.client.zgw.ztc.model.generated.ZaakType
+import nl.info.zac.authentication.LoggedInUser
+import nl.info.zac.enkelvoudiginformatieobject.EnkelvoudigInformatieObjectLockService
+import nl.info.zac.enkelvoudiginformatieobject.model.EnkelvoudigInformatieObjectLock
+import nl.info.zac.enkelvoudiginformatieobject.util.isSigned
+import nl.info.zac.policy.exception.PolicyException
+import nl.info.zac.policy.input.DocumentData
+import nl.info.zac.policy.input.DocumentInput
+import nl.info.zac.policy.input.TaakData
+import nl.info.zac.policy.input.TaakInput
+import nl.info.zac.policy.input.UserInput
+import nl.info.zac.policy.input.ZaakData
+import nl.info.zac.policy.input.ZaakInput
+import nl.info.zac.policy.output.DocumentRechten
+import nl.info.zac.policy.output.OverigeRechten
+import nl.info.zac.policy.output.TaakRechten
+import nl.info.zac.policy.output.WerklijstRechten
+import nl.info.zac.policy.output.ZaakRechten
+import nl.info.zac.search.model.DocumentIndicatie
+import nl.info.zac.search.model.ZaakIndicatie
+import nl.info.zac.search.model.zoekobject.DocumentZoekObject
+import nl.info.zac.search.model.zoekobject.TaakZoekObject
+import nl.info.zac.search.model.zoekobject.ZaakZoekObject
+import nl.info.zac.util.AllOpen
+import nl.info.zac.util.NoArgConstructor
+import org.eclipse.microprofile.rest.client.inject.RestClient
+import org.flowable.task.api.TaskInfo
+
+@ApplicationScoped
+@NoArgConstructor
+@AllOpen
+@Suppress("TooManyFunctions")
+class PolicyService @Inject constructor(
+    private val loggedInUserInstance: Instance<LoggedInUser>,
+    @RestClient private val evaluationClient: OpaEvaluationClient,
+    private val ztcClientService: ZtcClientService,
+    private val lockService: EnkelvoudigInformatieObjectLockService,
+    private val zrcClientService: ZrcClientService
+) {
+    fun readOverigeRechten(): OverigeRechten =
+        evaluationClient.readOverigeRechten(
+            RuleQuery(UserInput(loggedInUserInstance.get()))
+        ).getResult()
+
+    fun readZaakRechten(zaak: Zaak): ZaakRechten {
+        val zaakType = ztcClientService.readZaaktype(zaak.zaaktype)
+        return readZaakRechten(zaak, zaakType)
+    }
+
+    fun readZaakRechten(zaak: Zaak, zaaktype: ZaakType): ZaakRechten {
+        val zaakData = ZaakData().apply {
+            this.open = zaak.isOpen
+            this.zaaktype = zaaktype.getOmschrijving()
+            this.opgeschort = zaak.isOpgeschort
+            this.verlengd = zaak.isVerlengd
+            this.besloten = zaaktype.getBesluittypen()?.isNotEmpty() == true
+        }
+        var statusType: StatusType? = null
+        zaak.status?.let {
+            val status = zrcClientService.readStatus(it)
+            statusType = ztcClientService.readStatustype(status.getStatustype())
+        }
+        zaakData.intake = isIntake(statusType)
+        zaakData.heropend = isHeropend(statusType)
```

Comment:
This can be in the `apply` block too I believe

Reply:
true, good one. was on my TODO list but then I forgot

Reply:
fixed it now